DigitalLearningSolutions.Data/Helpers/LearningHubSsoSecurityHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace DigitalLearningSolutions.Data.Helpers
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using DigitalLearningSolutions.Data.Services;
+
+    public class LearningHubSsoSecurityHelper
+    {
+        private static readonly DateTime UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        private readonly IClockService clockService;
+
+        public LearningHubSsoSecurityHelper(IClockService clockService)
+        {
+            this.clockService = clockService;
+        }
+
+        public string GenerateHash(string state, string secretKey)
+        {
+            var secondsSinceEpoch = GetSecondsSinceEpoch();
+            var encoder = new UTF8Encoding();
+            var salt = encoder.GetBytes(secretKey);
+            var pwd = encoder.GetBytes(state + secondsSinceEpoch);
+            return GetHash(pwd, salt);
+        }
+
+        public bool VerifyHash(string state, string secretKey, string hash)
+        {
+            var secondsSinceEpoch = GetSecondsSinceEpoch();
+            var encoder = new UTF8Encoding();
+            var salt = encoder.GetBytes(secretKey);
+            var toleranceInSec = 60;

```

Comment:
The documentation is explicit on this tolerance so this is probably ok. But I'm wondering whether it is worth this living in config - but then actually within this context that would mean a parameter? Just in ase they change the tolerance level.

Reply:
I'm down with putting it in config aye.